### PR TITLE
EIP-3541 london test fixes

### PR DESCRIPTION
**Pull request by fselmo**
### What was wrong?

- LondonComputation was raising an error where it should set the error and return the computation. This fixes broken ethereum tests related to this new logic from EIP-3541.
- Ethereum tests released a new version (v10.0)

### How was it fixed?

- Set the `computation.error` and return the computation rather than raise the VMError
- Updated ethereum tests submodule to new version v10.0

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [X] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/py-evm/blob/master/newsfragments/README.md)

[//]: # (See: https://py-evm.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/py-evm/blob/master/newsfragments/README.md)

#### Cute Animal Picture

IMG_20210915_165945

### CODE REVIEW

File: tests/core/vm/test_london.py
Comment (carver) [inline tests/core/vm/test_london.py:169]: I kind of liked the clarification you had in there before, like:

```suggestion
    # assert gas is consumed, but not all gas. This falls within a range:
```

File: newsfragments/2029.bugfix.rst
Comment (carver) [inline newsfragments/2029.bugfix.rst:1]: When fixing a bug that has not yet been released, it's best to not include the release note. This explanation makes sense to go in the PR, but doesn't need to be surfaced to users.

**Comment by fselmo** [comment]
@carver @kclowes, these changes should resolve the failing tests on #2021 related to the reserved bytecode (EIP-3541).

**Review by carver — APPROVED** [review]
Cool, all of the refactor & fix in the spurious_dragon computation looks like an improvement to me. Yay for fewer branches.

The big glaring issue is: no cute animal picture! X)

**Comment by fselmo** [comment]
> The big glaring issue is: no cute animal picture! X)

I'm a monster... how could I forget 😞

**Comment by fselmo** [comment]
@kclowes @carver updated. I am good to merge this if you are. No write permissions on my end.

**Comment by carver** [comment]
Also just added you for write access @fselmo

**Pull request merged**
